### DISCUSSION TITLE
impoveDefenses:  Upgrade vs Add

### DISCUSSION THREAD

**Discussion by Bluntaxe** [League-of-Fabulous-Developers/FoundryVTT-Fabula-Ultima#144] [Show and tell]
I was setting up some skills and effect using ProjectFUTextCommands

I needed an effect that increased Magic Defense by 1.  When you add an effect directly to a character you can select to change the mode from Add/Upgrade/Downgrade etc.

However in the **Guided Effect Creation** selecting **Improve Defenses**
it doesn't give you those options; it just says 'Improve to".  Putting in '1' I noticed when I applied the effect to a character it no longer showed a value for the magic defense. 

Digging through the code - I changed in the file inline-effects.mjs
under improveDefenses
I changed
mode: CONST.ACTIVE_EFFECT_MODES.UPGRADE
to
mode: CONST.ACTIVE_EFFECT_MODES.ADD

this seems to make the effect work the way I need.  I'm not sure if this will mess something else up elsewhere.

**Comment by spyrella** [comment]
Yup, this is correct, and in the latest version, it uses add for the change mode.
